utest/shared/src/test/scala/test/utest/RetryTests.scala

```diff
@@ -47,3 +66,32 @@ object LocalRetryTests extends utest.TestSuite{
   }
 }
 
+// TODO: bug with retry assert
+// object LocalRetryBeforeEachTests extends utest.TestSuite{
+//   var x = 0
+//   override def utestBeforeEach() = {
+//     println(s">>> before each x: $x")
+//     x = 0
+//   }
+//   val flaky = new FlakyThing
+
+//   def tests = Tests{
+//     'hello - retry(3){
+//       println(s">>> x: $x")
+//       assert(x == 0)
+//       x += 1
+//       flaky.run
+//     }
+//   }
+// }
+
+// -------------------------------- Running Tests --------------------------------
+//   Setting up CustomFramework
+//   >>> before each x: 0
+//   >>> x: 0
+//   >>> x: 1
+//   >>> x: 1
+//   >>> x: 1
+// X test.utest.LocalRetryBeforeEachTests.hello 20ms
+// utest.AssertionError: assert(x == 0)
+// utest.asserts.Asserts$.assertImpl(Asserts.scala:114)
```

Comment:
I'm not sure if this is a bug, as the local retry runs in a different context as the Executor, 
in this case the `utestBeforeEach` is called once.
It might need just a clarification in the docs

Reply:
No, this is fine. Local `retry(n)`s are standalone things, totally independent of whatever `utestWrap` is doing